### PR TITLE
Implement ConvertSubnetTx

### PR DESCRIPTION
**Pull request by ruijialin-avalabs**
This PR adds `ConvertSubnetTx` to pvm etna builder.
- added `ConvertSubnetTx` to pvm codec at index 35
- added serializable object `ConvertSubnetTx` and `ConvertSubnetValidator`
- moved `getMemoComplexity` to pvm/txs/fee/complexity.ts and renamed it to `getBytesComplexity`.
- added `getSubnetValidatorsComplexity` and `convertSubnetTx` to pvm/txs/fee/complexity.ts
- added `INTRINSIC_CONVERT_SUBNET_VALIDATOR_COMPLEXITIES` and `INTRINSIC_CONVERT_SUBNET_TX_COMPLEXITIES`
- added `isConvertSubnetTx` typeguard
- add `NODE_ID`, `BLS_SIGNATURE`, and `BLS_PUBLIC_KEY` to getEnvVars.  which gets the value from info.getNodeId for the network
- added examples to `createSubnetTx`, `createChainTx`, `convertSubnetTx` 
   -`createSubnetTx` and `createChainTx` have to be issued first before `convertSubnetTx` can be issued.

- Successful txs in etna explorer for address `P-custom1s4k9fch6uyhvv7necq070nzljgrqvazkpgles6`:
  1. [createSubnetTx](https://feat-etna-devnet.subnets.pages.dev/p-chain/tx/2qgPgFfLHad5BwiEB9cqSEv4kFMPicp7BWoKDS3BhpFgDhe6Ek)
  2. [createChainTx](https://feat-etna-devnet.subnets.pages.dev/p-chain/tx/2sT1n9G1dvALJeyxsStDhZS86HV7vYFC2GGB7nVz99BxRKZDxe)
  3. [convertSubnetTx](https://feat-etna-devnet.subnets.pages.dev/p-chain/tx/2c6n6wrSa55arszPQsph6xoZdDoUo4WJjke9YB8aD3UkoPHxkm)

### CODE REVIEW

File: src/vms/pvm/etna-builder/builder.ts [hunk 1]
```diff
@@ -1327,3 +1321,108 @@ export const newTransferSubnetOwnershipTx: TxBuilderFn<
...
+
+/**
+ * Creates a new unsigned PVM convert subnet transaction
+ * (`NewConvertSubnetTx`) using calculated dynamic fees.
+ *
+ * @param props {NewConvertSubnetTxProps}
+ * @param context {Context}
+ * @returns {UnsignedTx} An UnsignedTx.
+ */
+export const newConvertSubnetTx: TxBuilderFn<NewConvertSubnetTxProps> = (
```
Comment (erictaylor) [inline src/vms/pvm/etna-builder/builder.ts:1356]: Let's add test coverage for this builder.

File: src/serializable/primitives/bytes.ts [hunk 2]
```diff
@@ -34,6 +34,10 @@ export class Bytes extends Primitives {
     return concatBytes(bytesForInt(this.bytes.length), this.bytes);
   }
 
+  toBytesWithoutLength() {
+    return this.bytes;
+  }
+
```
Comment (erictaylor) [inline src/serializable/primitives/bytes.ts:40]: should this be `value()`? That would seem to better align with other primitives maybe.

**Review by erictaylor — CHANGES_REQUESTED** [review]
Looking good.

Left some initial comments. Let's make sure we are adding test coverage.

**Comment by rictorlome** [comment]
Thank you! Also, could you please add the new builder function to the etna-examples folder? Feel free to add `createSubnet` too, so we can verify this. Also, please link to a successful tx in the etna explorer, to confirm the network accepts this tx. Thanks

File: src/vms/pvm/etna-builder/builder.ts (hunk 1 above)
Comment (ruijialin-avalabs) [reply src/vms/pvm/etna-builder/builder.ts:1362]: added [here](dab22dd79c8a40f9e85684aa8dae1612c421401f)

File: src/vms/pvm/etna-builder/builder.ts [hunk 3]
```diff
@@ -65,9 +70,13 @@ import {
   getOutputComplexity,
   getOwnerComplexity,
   getSignerComplexity,
+  getBytesComplexity,
+  getConvertSubnetValidatorsComplexity,
 } from '../txs/fee';
 import { spend } from './spend';
 import { useSpendableLockedUTXOs, useUnlockedUTXOs } from './spend-reducers';
+import type { ConvertSubnetValidator } from '../../../serializable/fxs/pvm/convertSubnetValidator';
+import { INTRINSIC_CONVERT_SUBNET_TX_COMPLEXITIES } from '../txs/fee/constants';
```
Comment (erictaylor) [inline src/vms/pvm/etna-builder/builder.ts:79]: We should import this from `../txs/fee` like the others above.

File: src/vms/pvm/etna-builder/builder.ts [hunk 4]
```diff
@@ -1327,3 +1327,123 @@ export const newTransferSubnetOwnershipTx: TxBuilderFn<
...
+  validators: ConvertSubnetValidator[];
+  /**
+   * Indices of existing subnet owners.
+   */
+  subnetAuth: readonly number[];
+}>;
+
+/**
+ * Creates a new unsigned PVM convert subnet transaction
+ * (`NewConvertSubnetTx`) using calculated dynamic fees.
```
Comment (erictaylor) [inline src/vms/pvm/etna-builder/builder.ts:1356]: This should say `ConvertSubnetTx` not `NewConvertSubnetTx`.

File: src/vms/pvm/etna-builder/builder.test.ts [hunk 5]
```diff
@@ -1007,4 +1012,103 @@ describe('./src/vms/pvm/etna-builder/builder.test.ts', () => {
...
+          chainId: 'h5vH4Zz53MTN2jf72axZCfo1VbG1cMR6giR4Ra2TTpEmqxDWB',
+        },
+        testContext,
+      );
+    } catch (error) {
+      expect((error as Error).message).toEqual(
+        'Validator weight must be greater than 0',
+      );
+    }
+  });
```
Comment (erictaylor) [inline src/vms/pvm/etna-builder/builder.test.ts:1113]: Unnecessary describe.each block. Can we move this test to be nested in `describe('ConvertSubnetTx')`

**Review by erictaylor — CHANGES_REQUESTED** [review]
Looks great. Couple small cleanup requests.

**Review by erictaylor — APPROVED** [review]

File: examples/p-chain/import.ts [hunk 6]
```diff
@@ -4,17 +4,17 @@ import { getContextFromURI } from '../../src/vms/context';
...
-const P_CHAIN_ADDRESS = process.env.P_CHAIN_ADDRESS;
-const X_CHAIN_ADDRESS = process.env.X_CHAIN_ADDRESS;
-const PRIVATE_KEY = process.env.PRIVATE_KEY;
+import { getEnvVars } from '../utils/getEnvVars';
 
 const main = async () => {
+  const { AVAX_PUBLIC_URL, P_CHAIN_ADDRESS, PRIVATE_KEY, X_CHAIN_ADDRESS } =
+    getEnvVars();
+
   if (!P_CHAIN_ADDRESS || !X_CHAIN_ADDRESS || !PRIVATE_KEY) {
```
Comment (rictorlome) [inline examples/p-chain/import.ts:13]: You don't need to throw these errors anymore because `getEnvVars` does it.

**Review by rictorlome — COMMENTED** [review]
Thank you. Can this be added to the burn checker as well?

**Review by rictorlome — APPROVED** [review]

**Pull request merged**